palidintheonly/MonkeyBytes-API
# Tuesday api updates

Pull Request: Enhancements to Imagery, Avatars, and Discord Notifications

Overview:
This pull request introduces a series of enhancements and optimizations to the MonkeyBytes-API, including updates to pigeon images, a transition to RoboHash for avatar generation, and improvements to Discord notification formatting.

Key Changes:

1. Updated Pigeon Images:
   - Replaced the old pigeon image list with ten new vibrant illustrations sourced from i.ibb.co. These images are now served via the /testing endpoint, providing users with a refreshed visual experience.

2. Avatar Generation via RoboHash:
   - Switched the avatar generation service from DiceBear to RoboHash. The new avatars offer a distinctive robotic theme, ensuring each profile picture is unique and visually engaging.

3. Improved Discord Time Formatting:
   - Updated the timestamp formatting in Discord webhook notifications from 24-hour to a 12-hour format. This change enhances readability and aligns with common time-telling practices for users receiving notifications.

4. Codebase Streamlining:
   - Removed redundant code segments, including duplicate import statements, and pruned unnecessary lines for better maintainability and efficiency.

Additional Improvements:

- Logging and Error Handling: Refined the logging process using Winston to ensure all operations are efficiently tracked, and errors are captured with greater accuracy.
- System Stability: Code optimizations were performed to improve performance and ensure the API runs smoothly without interruptions.

How to Test:

1. Pigeon Images:
   - Access the /testing endpoint and verify that the new pigeon images are being returned as expected.

2. Avatars:
   - Check that profile pictures are generated via RoboHash by querying the /testing endpoint and inspecting the avatars.

3. Discord Webhook Notifications:
   - Monitor Discord webhook messages to ensure that timestamps are displayed in a 12-hour format.

Summary:
This update enhances both the functionality and the visual experience of the MonkeyBytes-API. The changes improve user interaction with fresher visuals and optimized time formatting, all while streamlining the backend for better performance.
